src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -109,9 +113,6 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
                         case ExternMethodAccessorNode accessorNode:
                             objectWriter.GetCodeForExternMethodAccessor(accessorNode);
                             continue;
-                        case ModulesSectionNode modulesSectionNode:
```

Comment:
Is this just for multifile compilation?

Reply:
Well, sort of. There is always one such node produced in a compilation (single-file too).

Reply:
but we don't need it anymore because we are not using `RtRHeaderWrapper` ?

Reply:
It is still needed and still generated. The only change is how the runtime gets to it - it uses the linked-generated symbols instead of `RtRHeaderWrapper`.

After the linker runs, the results end up looking something like this:
```
__start__modules
__modules (section with this node as the only member)
__stop__modules
```

Reply:
oh, I think I see, its emitted now in `EmitKeepAliveList`